src/nnet3/nnet-simple-component.h

```diff
@@ -873,6 +875,83 @@ class PerElementScaleComponent: public UpdatableComponent {
 };
 
 
+// PerElementOffsetComponent offsets each dimension of its input with a separate
+// trainable bias; it's like an affine component with fixed weight matrix which is always equal to I.
+class PerElementOffsetComponent: public UpdatableComponent {
+ public:
+  virtual int32 InputDim() const { return offsets_.Dim(); }
+  virtual int32 OutputDim() const { return offsets_.Dim(); }
+
+  virtual std::string Info() const;
+  virtual void InitFromConfig(ConfigLine *cfl);
+
+  PerElementOffsetComponent() { } // use Init to really initialize.
+  virtual std::string Type() const { return "PerElementOffsetComponent"; }
+  virtual int32 Properties() const {
```

Comment:
Yes this component is affine in both parameters and inputs but the kLinearInParameters bit is also set for AffineComponent which is not linear in all parameters (i.e. in bias parameters). Is that right?